spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/XsuaaJwtDecoder.java

```diff
@@ -123,10 +125,26 @@ private Jwt verifyToken(JWT jwt) {
 		}
 	}
 
-	private Jwt verifyToken(String token, String jku, String kid, String uaaDomain) {
+	@Nullable
+	private static String getZid(JWT jwt) {
+		String zid;
+		try {
+			zid = jwt.getJWTClaimsSet().getStringClaim(
+					TokenClaims.CLAIM_ZONE_ID);
+
+		} catch (ParseException e) {
+			zid =null;
+		}
+		if (zid != null && zid.isBlank()){
+			zid = null;
+		}
+		return zid;
+	}
+
+	private Jwt verifyToken(String token, String jku, String kid, String uaaDomain, String zid) {
 		try {
 			canVerifyWithKey(jku, kid, uaaDomain);
```

Comment:
remove jku method parameter and use composeJku(uaaDomain, zid) instead? Then only pass jku and kid to canVerifyWithKey?